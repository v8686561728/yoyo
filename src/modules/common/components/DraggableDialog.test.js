import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { DraggableDialog, PaperComponent } from "./DraggableDialog";

let data = {
  values: {
    email: "ashish@gmail.com"
  },
  values2: {
    email: "ashish@.com"
  },
  values3: {
    email: ""
  },
  errors: "",
  touched: false,
  handleChange: jest.fn(),
  handleBlur: jest.fn(),
  handleSubmit: jest.fn(),
  isSubmitting: jest.fn()
};
let props = {
  classes: {
    root: "root"
  },
  getEmail: jest.fn()
};
it("should render DraggableDialog", () => {
  let wrapper = shallow(<DraggableDialog {...props} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should trigger handleOpenClose click event", () => {
  let wrapper = shallow(<DraggableDialog {...props} />);
  wrapper.find("#sendGift").simulate("click");
});

it("should render PaperComponent", () => {
  let wrapper = shallow(<PaperComponent />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("should trigger handleSendAndClose click event", () => {
  let wrapper = shallow(<DraggableDialog {...props} values={data.values} />);
  shallowToJson(wrapper).children[1].children[1].children[1].props.validate(
    data.values
  );
  shallowToJson(wrapper).children[1].children[1].children[1].props.validate(
    data.values2
  );
  shallowToJson(wrapper).children[1].children[1].children[1].props.validate(
    data.values3
  );
  shallowToJson(wrapper).children[1].children[1].children[1].props.onSubmit(
    data.values
  );
});
  
 
 
  