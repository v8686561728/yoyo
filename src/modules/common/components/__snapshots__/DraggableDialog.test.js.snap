// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DraggableDialog 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    color="primary"
    id="sendGift"
    onClick={[Function]}
    variant="contained"
  >
    send this gift card
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Dialog))
    PaperComponent={[Function]}
    aria-labelledby="draggable-dialog-title"
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="draggable-dialog-title"
      style={
        Object {
          "cursor": "move",
        }
      }
    >
      Send Gift card
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogContentText))>
        You can send this gift card to anyone. Just fill-in the email address below
      </WithStyles(ForwardRef(DialogContentText))>
      <Formik
        enableReinitialize={false}
        initialValues={
          Object {
            "email": "",
          }
        }
        isInitialValid={false}
        onSubmit={[Function]}
        validate={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
      >
        <Component />
      </Formik>
    </WithStyles(ForwardRef(DialogContent))>
  </WithStyles(ForwardRef(Dialog))>
</Fragment>
`;

exports[`should render PaperComponent 1`] = `
<Draggable
  allowAnyClick={false}
  axis="both"
  bounds={false}
  cancel="[class*=\\"MuiDialogContent-root\\"]"
  defaultClassName="react-draggable"
  defaultClassNameDragged="react-draggable-dragged"
  defaultClassNameDragging="react-draggable-dragging"
  defaultPosition={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  disabled={false}
  enableUserSelectHack={true}
  grid={null}
  handle={null}
  offsetParent={null}
  onDrag={[Function]}
  onMouseDown={[Function]}
  onStart={[Function]}
  onStop={[Function]}
  position={null}
  scale={1}
  transform={null}
>
  <WithStyles(ForwardRef(Paper)) />
</Draggable>
`;
